apm-sniffer/apm-sdk-plugin/spymemcached-2.x-plugin/src/main/java/org/skywalking/apm/plugin/spymemcached/v2/MemcachedMethodInterceptor.java

```diff
@@ -0,0 +1,45 @@
+package org.skywalking.apm.plugin.spymemcached.v2;
+
+import java.lang.reflect.Method;
+
+import org.skywalking.apm.agent.core.context.ContextManager;
+import org.skywalking.apm.agent.core.context.tag.Tags;
+import org.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.skywalking.apm.network.trace.component.ComponentsDefine;
+
+public class MemcachedMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+    private static final String SPY_MEMCACHE = "SpyMemcached/";
+
+    @Override 
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        Object[] arguments = allArguments;
+        String peer = String.valueOf(objInst.getSkyWalkingDynamicField());
+        AbstractSpan span = ContextManager.createExitSpan(SPY_MEMCACHE + method.getName(), peer);
+        span.setComponent(ComponentsDefine.MEMCACHE);
+        Tags.DB_TYPE.set(span, ComponentsDefine.MEMCACHE.getName());
+        SpanLayer.asDB(span);
+        if (allArguments.length > 0 && arguments[0] instanceof String) {
```

Comment:
Is there any chance, allArguments.length !=0 or allArguments[0] is not a String?  Redis plugin did this because there are many methods using the same interceptor. Are you doing the same thing?

Reply:
If so, what are the db statements for them?